web/src/engine/osk/src/visualKeyboard.ts

```diff
@@ -489,8 +489,8 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
           if(previewHost) {
             this.gesturePreviewHost = previewHost;
             trackingEntry.previewHost = previewHost;
-            sourceTrackingMap[source.identifier].key = key;
           }
+          sourceTrackingMap[source.identifier].key = key;
```

Comment:
Note line 481 - we need to update the key we just highlighted even if we don't show a preview for it.  That's the only change needed.